components/ui/common/form/datePicker/index.tsx

```
@@ -67,20 +67,32 @@ export default function DatePickerComponent({
   const [startDate, setStartDate] = useState<Date>(startingDate);
   const [endDate, setEndDate] = useState<Date>(endingDate);
 
+  // Ensure endDate is always after startDate and propagate changes
   useEffect(() => {
-    if (startDate > endDate)
-      setEndDate(new Date(startDate.getTime() + 60 * 60 * 1000));
-  }, [startDate, endDate]);
+    if (startDate > endDate) {
+      const newEndDate = new Date(startDate.getTime() + 60 * 60 * 1000);
+      setEndDate(newEndDate);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/components/ui/common/form/datePicker/index.tsx:74:7
  72 |     if (startDate > endDate) {
  73 |       const newEndDate = new Date(startDate.getTime() + 60 * 60 * 1000);
> 74 |       setEndDate(newEndDate);
     |       ^^^^^^^^^^ Avoid calling setState() directly within an effect
  75 |       onChange("endDate", toISOString(newEndDate));
  76 |     }
  77 |   }, [startDate, endDate, onChange]);

(react-hooks/set-state-in-effect)